modules/serverless-create-traffic-split-kn.adoc

```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// * serverless/develop/serverless-traffic-management.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-create-traffic-split-kn_{context}"]
+= Creating a traffic split by using the Knative CLI
+
+Using the `kn` CLI to create traffic splits provides a more streamlined and intuitive user interface over modifying YAML files directly. You can use the `kn service update` command to split traffic between revisions of a service.
+
+.Prerequisites
+
+* The {ServerlessOperatorName} and Knative Serving are installed on your cluster.
+* You have installed the `kn` CLI.
+* You have created a Knative service.
+
+.Procedure
+
+* Specify the revision of your service and what percentage of traffic you want to route to it by using the `--traffic` tag with a standard `kn service update` command:
++
+.Example command
+[source,terminal]
+----
+$ kn service update <service_name> --traffic <revision>=<percentage>
+----
++
+Where:
++
+** `<service_name>` is the name of the Knative service that you are configuring traffic routing for.
+** `<revision>` is the revision that you want to configure to receive a percentage of traffic. You can either specify the name of the revision, or a tag that you assigned to the revision by using the `--tag` flag.
+** `<percentage>` is the percentage of traffic that you want to send to the specified revision.
+
+* Optional: The `--traffic` flag can be specified multiple times in one command. For example, if you have a revision tagged as `@latest` and a revision named `stable`, you can specify the percentage of traffic that you want to split to each revision as follows:
++
+.Example command
+[source,terminal]
+----
+$ kn service update example-service --traffic @latest=20 stable=80
```

Comment:
I'd suggest the following for this example to highlight the change from previous behavior.

If user is executing an update operation, it can be combined with only a % number without any additional context. In that case we infer the value is meant to be used for updated service (i.e new revision resulting from update op).

It's effectively combining a few step required previously.  

```suggestion
$ kn service update example-service --env VERSION=v2 --traffic 20
```

Reply:
With regard to comment below, the message above should be reworded (I'll try my best here :)) e.g.

```
 Optional: The `--traffic` flag can be specified multiple times in one command. For example, if you perform update operation resulting in creation of a new revision, you can specify the percentage of traffic that you want to allocate for this revision in the update command.
```

Reply:
+1 with @dsimansk 

Also, the values should be comma separated with the `--traffic` flag, like so:

`kn service update example-service --traffic @latest=20,stable=80`

Reply:
@dsimansk my only question is that, adding the `--env` flag without explaining it feels like we're actually adding a sort of "unknown" bit here and not really simplifying anything. Can you explain what this is for, and what the options are? E.g. to me it's not clear if only `VERSION` can be specified for this or it could also be the revision name, etc as previously?

Reply:
Ahh, sorry about that Ashleigh. The `--env` flag is used for environment variables. I've used it for the sake of demonstrating multipe steps of update operations in single one, i.e. modifying some other Knative Service property and `Traffic` together. Previously use would need to update env var, retrieve new revision name and then update traffic.

Reply:
@dsimansk can `--env` be left out of the example altogether then? I feel like it's not really necessary for these examples and is just widening the scope beyond showing how to use the traffic flag too much? Maybe a follow up Jira later to show use cases where we use traffic with multiple other flags rather than complicating this update to the functionality for now?

Reply:
Per our Slack chat my suggested command doesn't work without combination with change producing flag, e.g. `--env`. Because it needs either all the revisions or "change producing flag" to attach to newly created revision.

```
➜  ~ kn service update example-service --traffic 20                         
Error: unable to allocate the remaining traffic: 80%. Reason: incorrect number of revisions specified. Only 1 revision should be missing
```

Using the 100%, all revisions provided is absolutely adequate and sufficient here:
```
kn service update example-service --traffic @latest=20,stable=80
```



> Maybe a follow up Jira later to show use cases where we use traffic with multiple other flags rather than complicating this update to the functionality for now?

Finally, I'm fine with a follow-up approach. It's really more "advanced" scenario, that demonstrates more just `--traffic ` flag functionality. "Change producing flag" can be whatever modifies Service configuration to be replaced with new Revision, not necessarily just `--env` flag, but it's easy to demonstrate with it.

As an example:
```
kn service update test --env v=2
kn revision list
kn service update test --traffic @latest=20,stable=80

# We tried to get rid of intermediate step within update command, that'd simplified as
kn service update test --env VERSION=v2 --traffic 20
```